plugins/jobtype/src/azkaban/jobtype/SparkJobArg.java

```diff
@@ -0,0 +1,36 @@
+package azkaban.jobtype;
+
+public enum SparkJobArg {
+
+  MASTER("master", "--master", "yarn-cluster"), // just to trick the eclipse formatter
+  SPARK_JARS("jars", "--jars", "./lib/*"), //
+  EXECUTION_JAR("execution.jar", null, null), //
+  EXECUTION_CLASS("execution.class", "--class", null), //
+  NUM_EXECUTORS("num.executors", "--num-executors", "2"), //
+  EXECUTOR_CORES("executor.cores", "--executor-cores", "1"), //
+  QUEUE("queue", "--queue", "marathon"), //
+  DRIVER_MEMORY("driver.memory", "--driver-memory", "2g"), //
+  EXECUTOR_MEMORY("executor.memory", "--executor-memory", "1g"), //
+  PARAMS("params", "null", ""), //
+  SPARK_CONF_PREFIX("spark.conf.", "--conf", ""), //
+  SPARK_DRIVER_PREFIX("spark.driver.", "--", ""), //
+  DRIVER_JAVA_OPTIONS("spark.driver.driver-java-options", "--driver-java-options", ""),
+  SPARK_FLAG_PREFIX("spark.flag.", "--", ""); // --help, --verbose, --supervise, --version
```

Comment:
@Victsm
Thanks for your comment!

I had 2 thoughts in mind when doing this part:
1. I'll provide a short hand for the commonly used specifications (things such as num.executors, exectuor.cores, queue, etc).  Obviously "commonly used" is not an exact term and we can discuss where the line falls.
2. I'll provide a way for people to specify everything spark supports (hence spark.conf.*, spark.driver.*, spark.flag.*)

So question for @Victsm 
1. which commands did you have in mind? Are they not expressable? Or they are just promoted as top level and you think they should?
2. Do you have the spark submit argument spec page?  I presume there is a page out there but the best I can find is http://spark.apache.org/docs/1.4.0/configuration.html, which is a SparkConf list and not specific to SparkSubmit.

Thanks!
John